core/property/registry.go

```diff
@@ -2,31 +2,66 @@ package property
 
 import (
 	"fmt"
-	"sync"
-
+	"github.com/TIBCOSoftware/flogo-lib/config"
 	"github.com/TIBCOSoftware/flogo-lib/core/data"
 	"github.com/TIBCOSoftware/flogo-lib/logger"
+	"reflect"
+	"regexp"
+	"sync"
 )
 
 var (
-	props = make(map[string]string)
-	mut   = sync.RWMutex{}
+	props    = make(map[string]interface{})
+	mut      = sync.RWMutex{}
+	regex    = regexp.MustCompile(config.GetPropertyDelimiterFormat())
+	resolver PropertyValueResolver
 )
 
+func init() {
+	logger.Debugf("Registering environment variable value resolver")
+	RegisterValueResolver(&data.EnvVarResolver{})
+}
+
+// Resolve value sourced from Enviornment variable or any other configuration management services
+type PropertyValueResolver interface {
+	//Resolve value for given name
+	Resolve(name string) interface{}
+}
+
 // Get returns the value of the property for the given id
 // If it is an environment property (for example {MY_PROP})
 // The value will be looked up in the os environment
-func Get(id string) string {
+func Get(id string) interface{} {
```

Comment:
As I said above, its not consistent behavior. IMHO, as a framework we should put less burden on the extension developers especially for type conversions.